src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -107,6 +108,26 @@ def _build_general_name(backend, gn):
         return x509.DirectoryName(
             _build_x509_name(backend, gn.d.directoryName)
         )
+    elif gn.type == backend._lib.GEN_EMAIL:
+        data = backend._ffi.buffer(
+            gn.d.rfc822Name.data, gn.d.rfc822Name.length
+        )[:].decode("ascii")
+        name, address = parseaddr(data)
+        parts = address.split("@")
```

Comment:
This should really be a `u"@"`, in practice it works fine this way, but I think it's slightly easier to read when it's explicitly unicode